src/main/java/me/xmrvizzy/skyblocker/utils/title/TitleContainer.java

```diff
@@ -0,0 +1,65 @@
+package me.xmrvizzy.skyblocker.utils.title;
+
+import me.xmrvizzy.skyblocker.SkyblockerMod;
+import net.fabricmc.fabric.api.client.command.v2.ClientCommandRegistrationCallback;
+import net.fabricmc.fabric.api.client.rendering.v1.HudRenderCallback;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.font.TextRenderer;
+import net.minecraft.client.gui.DrawContext;
+import net.minecraft.text.Text;
+import net.minecraft.util.Formatting;
+import net.minecraft.util.math.MathHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static net.fabricmc.fabric.api.client.command.v2.ClientCommandManager.literal;
+
+public class TitleContainer {
+    public static List<Title> titles = new ArrayList<>();
+    private static final Logger LOGGER = LoggerFactory.getLogger("skyblocker");
+    public static void init() {
+        HudRenderCallback.EVENT.register(TitleContainer::draw);
+    }
+    public static void addTitle(Title title)
+    {
+        title.active = true;
+        title.lastX = 0;
+        titles.add(title);
+    }
+    public static void draw(DrawContext context, float tickDelta) {
+        var client = MinecraftClient.getInstance();
+        TextRenderer textRenderer = client.textRenderer;
+
+        List<Title> toRemove = new ArrayList<>();
+
+        float x;
+        float width = 0;
+        context.getMatrices().push();
+        context.getMatrices().scale(3.0F, 3.0F, 3.0F);
+        for (Title title : titles) {
+            width += textRenderer.getWidth(title.text) * 3F + 10;
+        }
+        context.getMatrices().pop();
```

Comment:
These matrix transformations are unnecessary and should be removed since nothing is being rendered or affected by them.